libsolidity/interface/ErrorReporter.cpp

```diff
@@ -80,6 +84,16 @@ void ErrorReporter::error(Error::Type _type, SourceLocation const& _location, Se
 	m_errorList.push_back(err);
 }
 
+void ErrorReporter::abortIfExcessive()
+{
+	if (m_errorList.size() > 256)
+	{
+		auto err = make_shared<Error>(Error::Type::Warning);
```

Comment:
This cannot be a warning, because it might be that we have 256 warnings and errors appear only later.

Reply:
Not happy with the type, but what is the best type for it?

Reply:
Actually I was wondering if this should exclude (ignore) warnings when calculating the limit.

Reply:
It seems that (ignoring warnings) is the way to go:

```
,truffle/Assert.sol:640:43: Warning: Function declared as view, but this expression (potentially) modifies the state and thus requires non-payable (the default) or payable.
            _report(result, _appendTagged(_tag(A, "Tested"), _tag(B, "Against"), message));
                                          ^---------------^
,Warning: There are more than 256 errors. Aborting.

Error: Could not find artifacts for ./TestMath.sol from any sources
    at Resolver.require (/tmp/tmp.H6c4gxXTth/node_modules/truffle/build/cli.bundled.js:50099:9)
    at TestResolver.require (/tmp/tmp.H6c4gxXTth/node_modules/truffle/build/cli.bundled.js:96057:30)
    at /tmp/tmp.H6c4gxXTth/node_modules/truffle/build/cli.bundled.js:209768:30
    at Array.map (native)
    at /tmp/tmp.H6c4gxXTth/node_modules/truffle/build/cli.bundled.js:209766:33

  Command failed: truffle test --network testing
```